libs/payments/ui/src/lib/client/components/CheckoutCheckbox.tsx

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+'use client';
+
+import * as HoverCard from '@radix-ui/react-hover-card';
+import { useState } from 'react';
+
+interface CheckoutCheckboxProps {
+  isRequired: boolean;
+  termsOfService: string;
+  privacyNotice: string;
+  notifyCheckboxChange: (isChecked: boolean) => void;
+}
+
+export function CheckoutCheckbox({
+  isRequired,
+  termsOfService,
+  privacyNotice,
+  notifyCheckboxChange,
+}: CheckoutCheckboxProps) {
+  const [isChecked, setIsChecked] = useState(false);
+
+  const changeHandler = () => {
+    const newValue = !isChecked;
+    setIsChecked(newValue);
+    notifyCheckboxChange(newValue);
+  };
+
+  return (
+    <div className="my-6">
+      <HoverCard.Root open={isRequired && !isChecked}>
+        <label className="flex gap-5 items-center">
+          <HoverCard.Trigger>
+            <input
+              type="checkbox"
+              name="confirm"
+              className="grow-0 shrink-0 basis-4 scale-150 cursor-pointer"
+              checked={isChecked}
+              onChange={changeHandler}
+            />
+          </HoverCard.Trigger>
+          <span className="font-normal text-[13px] block">
+            I authorize Mozilla to charge my payment method for the amount
+            shown, according to{' '}
+            <a className="text-blue-500 underline" href={termsOfService}>
+              Terms of Service
+            </a>{' '}
+            and{' '}
+            <a className="text-blue-500 underline" href={privacyNotice}>
+              Privacy Notice
+            </a>
+            , until I cancel my subscription.
+          </span>
+        </label>
+        <HoverCard.Portal>
+          <HoverCard.Content
+            className="animate-slide-up z-[6]"
+            sideOffset={15}
+            align={'start'}
+            alignOffset={50}
+            arrowPadding={20}
+          >
+            <div className="text-white text-sm bg-alert-red py-1.5 px-4">
+              You need to complete this before moving forward
+            </div>
+            <HoverCard.Arrow
+              className="fill-alert-red left-[30px]"
```

Comment:
Not sure if this can/should be `left-7` (28px) or `left-8` (32px) [to avoid custom values]. Didn't see a 7.5 which would have hit that magic 30px sweet spot. https://tailwindcss.com/docs/top-right-bottom-left

Also, not sure if it should be `left-n` or `start-n` if this is rtl-aware. 🤷

Reply:
I think `fill-alert-red` is enough? The arrow didn't change in appearance when I edited/removed `left-[30px]`.